src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java

```diff
@@ -21,16 +21,19 @@
     public static final ParseField FIELD_INDEX = new ParseField("index", new String[0]);
     public static final ParseField FIELD_GEOSPATIAL = new ParseField("field", new String[0]);
     public static final ParseField FIELD_GEOSPATIAL_TYPE = new ParseField("type", new String[0]);
+    public static final ParseField FIELD_FEATURE_ID = new ParseField("feature_id", new String[0]);
     public static final ParseField FIELD_DATA = new ParseField("data", new String[0]);
     private final String indexName;
     private final String fieldName;
     private final String fieldType;
+    private final String featureId;
     private final Object data;
 
-    private UploadGeoJSONRequestContent(String indexName, String fieldName, String fieldType, Object data) {
+    private UploadGeoJSONRequestContent(String indexName, String fieldName, String fieldType, String featureId, Object data) {
         this.indexName = indexName;
         this.fieldName = fieldName;
         this.fieldType = fieldType;
+        this.featureId = featureId; // can be null
```

Comment:
In what case would this be null? Also, where will this be used?

Reply:
Feature can have field similar to "_id" for Document. In that case, we will use that value for Doc id while uploading